blob/blob.go

```diff
@@ -1073,3 +1093,9 @@ func wrapError(b driver.Bucket, err error) error {
 }
 
 var errClosed = gcerr.Newf(gcerr.FailedPrecondition, nil, "blob: Bucket has been closed")
+
+// PrefixedBucket returns a *Bucket based on b with all keys modified to have
+// prefix.
+func PrefixedBucket(b *Bucket, prefix string) *Bucket {
+	return NewBucket(driver.NewPrefixedBucket(b.b, prefix))
```

Comment:
Open question: should we allow/support providing a prefix that doesn't end in a "/"? I'm not sure what the use case is for a non-directory prefix, and it's easy to mess up (e.g,. `?prefix=mysubdir` instead of `prefix=mysubdir/`).

We could:
a) Document that `prefix` should likely end with a `/`.
b) Add a `/` to prefix if it doesn't have one (probably in `driver.NewPrefixedBucket`).
c) Change this and `driver.NewPrefixedBucket` to return an error if `prefix` doesn't end with a `/`.

I'm mildly leaning toward c), but let me know what you think.

Reply:
My preference is a), since I could imagine some other delimiter being used (like dashes)

Reply:
Updated docstring.